sklearn/feature_selection/tests/test_from_model.py

```diff
@@ -101,6 +103,48 @@ def test_feature_importances():
     mask = np.abs(transformer.estimator_.coef_) > 1e-5
     assert_array_equal(X_new, X[:, mask])
 
+    # Check with various values of k
+    max_n_features = X.shape[1]
+
+    transformer = SelectFromModel(estimator=est, max_n_features=0)
+    transformer.fit(X, y)
+    X_new = transformer.transform(X)
+    assert_equal(X_new.shape[1], 0)
+
+    transformer = SelectFromModel(estimator=est, max_n_features=max_n_features)
+    transformer.fit(X, y)
+    X_new = transformer.transform(X)
+    assert_equal(X_new.shape[1], max_n_features)
+
+    transformer = SelectFromModel(estimator=est, max_n_features='all')
+    transformer.fit(X, y)
+    X_new = transformer.transform(X)
+    assert_equal(X_new.shape[1], max_n_features)
+
+    transformer = SelectFromModel(estimator=est, max_n_features=5)
+    transformer.fit(X, y)
+    X_new = transformer.transform(X)
+    assert_equal(X_new.shape[1], 5)
+
+    transformer = SelectFromModel(estimator=est, max_n_features=-1)
```

Comment:
And put these tests in a function called `test_invalid_max_n_features()` (or something to that effect).